lib/OpenQA/Worker/Engines/isotovideo.pm

```diff
@@ -326,11 +340,28 @@ sub engine_workit {
         my $error = locate_local_assets(\%vars, $assetkeys);
         return $error if $error;
     }
+    my $casedir             = OpenQA::Utils::testcasedir($vars{DISTRI}, $vars{VERSION}, $shared_cache);
+    my $productdir          = OpenQA::Utils::productdir($vars{DISTRI}, $vars{VERSION}, $shared_cache);
+    my $relative_casedir    = substr($casedir,          length($vars{PRJDIR}) + 1);
+    my $relative_productdir = substr($productdir,       length($vars{PRJDIR}) + 1);
+    my $target_name         = substr($relative_casedir, 0, index($relative_casedir, '/'));
 
     $vars{ASSETDIR}   //= OpenQA::Utils::assetdir();
-    $vars{CASEDIR}    //= OpenQA::Utils::testcasedir($vars{DISTRI}, $vars{VERSION}, $shared_cache);
-    $vars{PRODUCTDIR} //= OpenQA::Utils::productdir($vars{DISTRI}, $vars{VERSION}, $shared_cache);
-
+    $vars{CASEDIR}    //= $relative_casedir;
+    $vars{PRODUCTDIR} //= $relative_productdir;
+
+    # if the CASEDIR is a git repository, we do not need to do symlink for CASEDIR.
+    # The code will be git clone in os-autoinst.
+    if ($vars{CASEDIR} !~ /http(s)?\:\/\//) {
+        my $link_casedir_error = _link_testrepo(catdir($vars{PRJDIR}, $target_name) , $pooldir, $target_name);
+        return $link_casedir_error if $link_casedir_error;
+    }
+    if ($vars{NEEDLES_DIR} && !File::Spec->file_name_is_absolute($vars{NEEDLES_DIR}) && $vars{NEEDLES_DIR} !~ /http(s)?\:\/\//)
```

Comment:
Maybe the check for a Git URL should be moved in a separate function so the regex is not repeated.

Reply:
According to https://github.com/os-autoinst/openQA/pull/3604#discussion_r544610157,  I removed the `if` block when doing symlink casedir, so there is only one.